src/platform/cc13x2_26x2/CC13X2_26X2Config.cpp

```diff
@@ -168,6 +268,63 @@ CHIP_ERROR CC13X2_26X2Config::WriteConfigValueStr(Key key, const char * str, siz
     return WriteConfigValueBin(key, (const uint8_t *) str, strLen);
 }
 
+CHIP_ERROR CC13X2_26X2Config::WriteKVS(const char * key, const void * value, size_t value_size)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint16_t subID;
+
+    if (FindKVSSubID(key, subID) == NVINTF_SUCCESS)
+    {
+        NVINTF_itemID_t val_item = CC13X2_26X2Config::kConfigKey_KVS_value.nvID;
+        // key already exists, update value
+        val_item.subID = subID;
+        VerifyOrExit(sNvoctpFps.updateItem(val_item, (uint16_t) value_size, (void *) value) == NVINTF_SUCCESS,
+                     err = CHIP_ERROR_PERSISTED_STORAGE_FAILED);
+    }
+    else
+    {
+        // key does not exist, likely case
+        intptr_t lock_key = sNvoctpFps.lockNV();
+
+        err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
+
+        NVINTF_itemID_t key_item = CC13X2_26X2Config::kConfigKey_KVS_key.nvID;
+        NVINTF_itemID_t val_item = CC13X2_26X2Config::kConfigKey_KVS_value.nvID;
+
+        /* Iterate through the subID range to find an unused subID in the
+         * keyspace.  SubID is a 10 bit value, reference
+         * `<simplelink_sdk>/source/ti/common/nv/nvocmp.c:MVOCMP_MAXSUBID`.
+         */
+        for (uint16_t i = 0; i < 0x3FF; i++)
+        {
+            key_item.subID = i;
+            if (sNvoctpFps.getItemLen(key_item) == 0U)
```

Comment:
Do we need to worry about valid zero-length values being stored and then treated as available slots?

Reply:
That would be a key of zero length. I can't store a key/value pair with a NULL key.

But yes for the underlying driver, writing a value of length zero is invalid. Do we have key/value pairs that would only be a key?

Reply:
Oh, good point about empty key, ok!

As far as values go: so far I think we don't have any empty values, based on a brief audit, but I don't see anything in the API that prevents them.  Maybe we should document something about values needing to be nonempty.

Reply:
The test for an empty value is an empty string `""` not a null value: https://github.com/project-chip/connectedhomeip/blob/master/examples/persistent-storage/KeyValueStorageTest.cpp#L53

And the Zephyr platform also rejects values of `0` length: https://github.com/project-chip/connectedhomeip/blob/master/src/platform/Zephyr/KeyValueStoreManagerImpl.cpp#L103

Reply:
Ah, OK, great.  :)